p2p/host/basic/basic_host.go

```diff
@@ -605,20 +586,13 @@ func (h *BasicHost) dialPeer(ctx context.Context, p peer.ID) error {
 		return err
 	}
 
-	// Clear protocols on connecting to new peer to avoid issues caused
-	// by misremembering protocols between reconnects
-	h.Peerstore().SetProtocols(p)
-
-	// identify the connection before returning.
-	done := make(chan struct{})
-	go func() {
-		h.ids.IdentifyConn(c)
-		close(done)
-	}()
-
-	// respect don contexteone
+	// TODO: Consider removing this? On one hand, it's nice because we can
+	// assume that things like the agent version are usually set when this
```

Comment:
@Stebalien Didn't we discuss that it's nice to block on `Identify` for outgoing connections so we can then directly check the peerstore for supported remote protocols, etc. ?

Why this TODO ?

Reply:
I did say that. I left this TODO because we may want to consider _not_ doing that.